app/controllers/plans_controller.rb

```diff
@@ -19,10 +19,13 @@ def enroll_in_meeting
     @course = Course.find(params[:course_id])
     @year = Year.find(params[:meeting_year])
     @term = Term.find(params[:meeting_term])
-    @session = Session.find(params[:meeting_session])
-    @meeting = Meeting.find_meeting(@course, @year, @term, @session)
+    @meeting = Meeting.find_meeting(@course, @year, @term)
     @plan = Plan.find(params[:plan][:id])
     if @meeting
+      if @plan.scheduled_classes.find_by(course_id: @meeting.course_id)
+        thesis_ids = [233, 234]
+        @plan.scheduled_classes.find_by(course_id: @meeting.course_id).enrollments.where(plan_id: @plan.id).delete_all
+      end
```

Comment:
Fix for thesis. Also Should be agnostic to other courses of the same name ebing scheduled.